doc/nrf/libraries/caf/buttons.rst

```diff
@@ -82,11 +82,37 @@ In this state, the module enables the GPIO interrupts and waits for the pin stat
 Whenever a button is pressed, the module switches to ``STATE_SCANNING``.
 When the switch occurs, the module submits a work with a delay set to :kconfig:`CONFIG_CAF_BUTTONS_DEBOUNCE_INTERVAL`.
 The work scans the keyboard matrix, or directly connected buttons (depends on configuration).
-If any button state change occurs, the module sends related event.
+If any button state change occurs, the module sends an event with the :c:member:`button_event.key_id` of that button.
 
 * If the button is kept pressed while the scanning is performed, the work will be resubmitted with a delay set to :kconfig:`CONFIG_CAF_BUTTONS_SCAN_INTERVAL`.
 * If no button is pressed, the module switches back to ``STATE_ACTIVE``.
 
+Key ID
+======
+
+The :c:member:`button_event.key_id` is an unique 2-byte value that depends on the module configuration described in the configuration file.
+
++------------------------------+
+| key_id                       |
++---------+---------+----+-----+
+| 0 .. 6  | 7 .. 13 | 14 | 15  |
++=========+=========+====+=====+
+| Row idx | Col idx | Reserved |
++---------+---------+----------+
+
+In this :c:member:`button_event.key_id` value:
+
+* The first seven least significant bits (``Row idx``) encodes the index of the ``row`` array that corresponds to the gpio port and the pin to which the button's row is connected.
+* The next seven bits (``Col idx``) encodes the index of the ``col`` array that corresponds to the gpio port and the pin to which the button's column is connected.
```

Comment:
You could mention that in case of buttons that are directly connected (no keyboard matrix), the buttons are denoted only by row and col is always set to zero.